_includes/tutorials/creating-first-apache-kafka-streams-application/kstreams/code/src/main/java/io/confluent/developer/Util.java

```diff
@@ -0,0 +1,97 @@
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.errors.TopicExistsException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.stream.Collectors;
+
+public class Util {
+
+    private final Logger logger = LoggerFactory.getLogger(Util.class);
+    private ExecutorService executorService = Executors.newFixedThreadPool(1);
+
+    public class Randomizer implements AutoCloseable, Runnable {
+        private Properties props;
+        private String topic;
+        private Producer<String, String> producer;
+        private boolean closed;
+
+        public Randomizer(Properties producerProps, String topic) {
+            this.closed = false;
+            this.topic = topic;
+            this.props = producerProps;
+            this.props.setProperty("client.id", "faker");
+        }
+
+        public void run() {
+            try (KafkaProducer producer = new KafkaProducer<String, String>(props)) {
+                Faker faker = new Faker();
+                while (!closed) {
+                    try {
+                        Object result = producer.send(new ProducerRecord<>(
+                                this.topic,
+                                faker.chuckNorris().fact())).get();
+                        Thread.sleep(5000);
+                    } catch (InterruptedException e) {
+                    }
+                }
+            } catch (Exception ex) {
+                logger.error(ex.toString());
+            }
+        }
+        public void close()  {
+            closed = true;
+        }
+    }
+
+    public Randomizer startNewRandomizer(Properties producerProps, String topic) {
+        Randomizer rv = new Randomizer(producerProps, topic);
+        executorService.submit(rv);
+        return rv;
+    }
+
+    public void createTopics(final Properties allProps, List<NewTopic> topics)
+            throws InterruptedException, ExecutionException, TimeoutException {
+        try (final AdminClient client = AdminClient.create(allProps)) {
+            logger.info("Creating topics");
+
+            client.createTopics(topics).values().forEach( (topic, future) -> {
+                try {
+                    future.get();
+                } catch (Exception ex) {
+                    logger.info(ex.toString());
+                }
+            });
+
+            Collection<String> topicNames = topics
+                .stream()
+                .map(t -> t.name())
+                .collect(Collectors.toCollection(LinkedList::new));
+
+            logger.info("Asking cluster for topic descriptions");
+            client
+                .describeTopics(topicNames)
+                .all()
+                .get(10, TimeUnit.SECONDS)
+                .forEach((name, description) -> logger.info("Topic Description: {}", description.toString()));
+        }
+    }
+
+    @Override
+    public void finalize() {
```

Comment:
`finalize` is [deprecated since java 9](https://docs.oracle.com/javase/10/docs/api/java/lang/Object.html#finalize()) (that was news to me) and it fails the `test java` phase of the build.

One suggestion is to make the `ExecutorService` a field on the `Randomizer` class and then you can close it in the `Randomizer#close` method.

Reply:
In 92e20858 I made Util autocloseable, and then changed the applications usage of it to be contained in a `try` block.

I didn't want to put `ExecutorService` inside the Randomizer because I wanted it's instance shared if there were more than one `Randomizer`. I can't make it static inside `Randomizer`, because that's not supported in Java 8 with inner classes. I also didn't want to break it out into it's own class to keep the tutorial simpler for the user.  I wanted to keep the number of files they have to create to a minimum.